custom/icds/views/custom_sms_report.py

```diff
@@ -8,23 +8,35 @@
 from custom.icds.forms import CustomSMSReportRequestForm
 from custom.icds.tasks.sms import send_custom_sms_report
 from custom.icds_core.const import SMSUsageReport_urlname
+from custom.icds.utils.custom_sms_report import CustomSMSReportTracker
 
 
 @method_decorator(toggles.ICDS_CUSTOM_SMS_REPORT.required_decorator(), name='dispatch')
 class SMSUsageReport(BaseMessagingSectionView):
     template_name = 'icds/sms/custom_sms_report.html'
     urlname = SMSUsageReport_urlname
     page_title = _('Custom SMS Usage Report')
+    report_tracker = CustomSMSReportTracker()
 
     @use_daterangepicker
     def dispatch(self, *args, **kwargs):
+        messages.info(self.request, _('Please note that this report takes a few hours to process.'))
         return super().dispatch(*args, **kwargs)
 
     @property
     def page_context(self):
+        reports_in_progress = self.report_tracker.active_reports
+        report_count = len(reports_in_progress)
+        disable_submit = True if report_count >= 3 else False
+        if disable_submit:
+            messages.info(self.request, 'Only 3 concurrent reports are allowed at a time.')
+        if report_count > 0:
+            display_message = _prepare_display_message(reports_in_progress, report_count)
+            messages.info(self.request, display_message)
```

Comment:
nit: most of this new block can be extracted in a method to `set_display_messages`.